WordPress/Classes/Utility/Media/CachedAsyncImage.swift

```diff
@@ -0,0 +1,127 @@
+    /// modifiers --- like ``Image/resizable(capInsets:resizingMode:)`` ---
+    /// to the loaded image, use the ``init(url:content:placeholder:)``
+    /// initializer instead.
+    ///
+    /// - Parameters:
+    ///   - url: The URL of the image to display.
+    init(url: URL?) where Content == Image {
+        self.init(url: url) { phase in
+            phase.image ?? Image(uiImage: .init())
+        }
+    }
+
+    /// Loads and displays a modifiable image from the specified URL using
+    /// a custom placeholder until the image loads.
+    ///
+    /// Until the image loads, SwiftUI displays the placeholder view that
+    /// you specify. When the load operation completes successfully, SwiftUI
+    /// updates the view to show content that you specify, which you
+    /// create using the loaded image. For example, you can show a green
+    /// placeholder, followed by a tiled version of the loaded image:
+    ///
+    ///     CachedAsyncImage(url: URL(string: "https://example.com/icon.png")) { image in
+    ///         image.resizable(resizingMode: .tile)
+    ///     } placeholder: {
+    ///         Color.green
+    ///     }
+    ///
+    /// If the load operation fails, SwiftUI continues to display the
+    /// placeholder. To be able to display a different view on a load error,
+    /// use the ``init(url:transaction:content:)`` initializer instead.
+    ///
+    /// - Parameters:
+    ///   - content: A closure that takes the loaded image as an input, and
+    ///     returns the view to show. You can return the image directly, or
+    ///     modify it as needed before returning it.
+    ///   - placeholder: A closure that returns the view to show until the
+    ///     load operation completes successfully.
+    init<I, P>(url: URL?, @ViewBuilder content: @escaping (Image) -> I, @ViewBuilder placeholder: @escaping () -> P) where Content == _ConditionalContent<I, P>, I: View, P: View {
+        self.init(url: url) { phase in
+            if let image = phase.image {
+                content(image)
+            } else {
+                placeholder()
+            }
+        }
+    }
+
+    init(
+        url: URL?,
+        transaction: Transaction = Transaction(),
+        imageDownloader: ImageDownloader = .shared,
+        @ViewBuilder content: @escaping (AsyncImagePhase) -> Content
+    ) {
+        self.url = url
+        self.transaction = transaction
+        self.imageDownloader = imageDownloader
+        self.content = content
+
+        self._phase = State(wrappedValue: .empty)
+        if let url, let image = cachedImage(from: url) {
+            self._phase = State(wrappedValue: .success(image))
+        }
+    }
+
+    @Sendable
+    private func load() async {
+        do {
+            if let url {
+                let image = try await Image(uiImage: imageDownloader.image(from: url))
+                if let cachedImage = cachedImage(from: url) {
```

Comment:
Exactly what I thought, but for a reason I couldn't understand, some images won't load when I do that.
Perhaps I nested the call under the wrong brace. I'll take another look.

Reply:
Resolved in https://github.com/wordpress-mobile/WordPress-iOS/commit/68f26790c11a3ecdafc53385b21b17f94dcac663 naturally since the check for `cachedImage` is removed